packages/gatsby/src/cache-dir/json-store.js

```diff
@@ -47,10 +42,9 @@ class JSONStore extends React.Component {
     }
   }
 
-  componentWillReceiveProps(nextProps) {
+  componentDidUpdate() {
```

Comment:
not sure about that change, I've used recomendation from https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops , but it probably means extra render